### PR TITLE
[UX] sync playsession every minute

### PR DESCRIPTION
**Pull request by BrettCleary**
# Summary
This PR syncs the play session every minute while playing a game.

closes https://github.com/HyperPlay-Gaming/product-management/issues/519

### CODE REVIEW

**Review by red-game-dev — APPROVED** [review]
LGTM 🚀

File: src/backend/main.ts [hunk 1]
```diff
@@ -1141,6 +1141,56 @@ ipcMain.on('logError', (e, err) => logError(err, LogPrefix.Frontend))
...
+
+  const totalPlaytime =
+    sessionPlaytimeInMinutes + BigInt(tsStore.get(`${appName}.totalPlayed`, 0))
+  tsStore.set(`${appName}.totalPlayed`, Number(totalPlaytime))
+
+  const game = gameManagerMap[runner].getGameInfo(appName)
+  const { gameIsEpicForwarderOnHP, hyperPlayListing } =
+    await gameIsEpicForwarderOnHyperPlay(game)
+
+  if (gameIsEpicForwarderOnHP && hyperPlayListing) {
```
Comment (jiyuu-jin) [inline src/backend/main.ts:1173]: Are we sending this for all Epic Forwarder games and all HyperPlay games even if they don't have a play session?

File: src/backend/main.ts [hunk 2]
```diff
@@ -1141,6 +1141,56 @@ ipcMain.on('logError', (e, err) => logError(err, LogPrefix.Frontend))
...
+    postPlaySessionTime(
+      hyperPlayListing.project_id,
+      parseInt((sessionPlaytimeInMs / BigInt(1000)).toString())
```
Comment (jiyuu-jin) [inline src/backend/main.ts:1176]: Couldn't you do something like `hyperPlayListing?.project_id || appName` and combine these lines.

File: src/frontend/components/UI/QuestsViewer/components/QuestDetailsWrapper/index.tsx [hunk 3]
```diff
@@ -178,6 +182,20 @@ export function QuestDetailsWrapper({
       loading: val.isLoading || val.isFetching
     })) ?? []
 
+  useEffect(() => {
```
Comment (jiyuu-jin) [inline src/frontend/components/UI/QuestsViewer/components/QuestDetailsWrapper/index.tsx:185]: Might be able to break these useEffects into hooks to make this a bit more readable/maintainable, maybe something like:

```ts
import { useEffect } from 'react';

// Custom hook for tracking events
function useTrackEvent(selectedQuestId: number | null) {
  useEffect(() => {
    if (selectedQuestId !== null) {
      window.api.trackEvent({
        event: 'Quest Viewed',
        properties: { quest: { id: selectedQuestId.toString() } },
      });
    }
  }, [selectedQuestId]);
}

// Custom hook for sync interval
function useSyncInterval(projectId: string, selectedQuestId: number | null, invalidateQuery: () => void) {
  useEffect(() => {
    const syncTimer = setInterval(async () => {
      await window.api.syncPlaySession(projectId, 'hyperplay');
      // allow for some time before read
      await wait(5000);
      await invalidateQuery();
      resetSessionStartedTime();
    }, 1000 * 60);

    return () => {
      clearInterval(syncTimer);
    };
  }, [projectId, selectedQuestId, invalidateQuery]);
}
```

**Review by jiyuu-jin — APPROVED** [review]

**Review by eliobricenov — APPROVED** [review]
code lgtm

File: src/backend/main.ts (hunk 1 above)
Comment (BrettCleary) [reply src/backend/main.ts:1173]: well. they will always have a play session by this point. I think you're asking if the game doesn't have an active play streak quest and yes that's correct. We send this every time a game is played. 

When we add the quest id's to the listings endpoint, we can add this info to the `GameInfo` object and then check before calling this from the 'launch' handler.

File: src/frontend/components/UI/QuestsViewer/components/QuestDetailsWrapper/index.tsx (hunk 3 above)
Comment (BrettCleary) [reply src/frontend/components/UI/QuestsViewer/components/QuestDetailsWrapper/index.tsx]: made the refactor 👍

File: src/backend/main.ts (hunk 2 above)
Comment (BrettCleary) [reply src/backend/main.ts]: yup. this simplifies the code a bit

**Review by nyghtstalker — APPROVED** [review]
It took a second after 100% for the 1/7 and bolt to appear but it worked!

image
image
image

**Pull request merged**